src-java/kilda-model/src/main/java/org/openkilda/model/HaFlowPath.java

```diff
@@ -80,25 +79,24 @@ private HaFlowPath() {
      */
     public HaFlowPath(@NonNull HaFlowPath entityToClone, HaFlow haFlow) {
         this();
-        data = HaFlowPathCloner.INSTANCE.deepCopy(entityToClone.getData(), haFlow, this);
+        data = HaFlowPathCloner.INSTANCE.deepCopy(entityToClone.getData(), haFlow);
     }
 
     @Builder
     public HaFlowPath(
             @NonNull PathId haPathId, @NonNull Switch sharedSwitch, SwitchId yPointSwitchId,
             FlowSegmentCookie cookie, MeterId sharedPointMeterId,
             MeterId yPointMeterId, GroupId yPointGroupId, long bandwidth, boolean ignoreBandwidth,
-            FlowPathStatus status, String sharedBandwidthGroupId, List<PathSegment> segments) {
+            FlowPathStatus status, List<FlowPath> subPaths) {
         data = HaFlowPathDataImpl.builder().haPathId(haPathId).sharedSwitch(sharedSwitch).yPointSwitchId(yPointSwitchId)
                 .cookie(cookie).yPointMeterId(yPointMeterId).sharedPointMeterId(sharedPointMeterId)
-                .yPointGroupId(yPointGroupId).bandwidth(bandwidth)
-                .ignoreBandwidth(ignoreBandwidth).status(status).sharedBandwidthGroupId(sharedBandwidthGroupId)
+                .yPointGroupId(yPointGroupId).bandwidth(bandwidth).ignoreBandwidth(ignoreBandwidth).status(status)
                 .build();
         // The reference is used to link sub flow edges back to the path. See {@link #setHaSubFlowEdges(Collection)}.
         ((HaFlowPathDataImpl) data).haFlowPath = this;
 
-        if (segments != null && !segments.isEmpty()) {
-            data.setSegments(segments);
+        if (subPaths != null && !subPaths.isEmpty()) {
```

Comment:
here we can use org.apache.commons.collections.CollectionUtils isEmpty that also checks if is null

Reply:
btw, can getSubPaths actually return a null?

Reply:
added CollectionUtils

subPaths has default value. But, in theory, you can path `null` into the builder method. I'm not sure what will happen in this case so added null check just in case